File: sdk/java/src/main/java/com/contextsolutions/securegateway/desktop/SecureGateway.java
package com.contextsolutions.securegateway.desktop;

import com.contextsolutions.securegateway.core.keystore.InMemoryKeyStore;

/**
 * Entry point for the desktop SDK — the single seam the host app toggles behind its relay
 * feature flag (PRD §8.3/§8.4). When the flag is off the app keeps its legacy local-sync
 * path; when on, it calls {@link #desktop(DesktopConfig)} and uses the returned client.
 * The QR payload is versioned ({@code v:1}), so a legacy QR (no {@code v}) routes to the
 * old behavior.
 */
public final class SecureGateway {

    private SecureGateway() {
    }

    public static DesktopClient desktop(DesktopConfig config) {
        if (config.keyStore == null) {
            config.keyStore = new InMemoryKeyStore();
        }
        return new DesktopClient(config);
    }
}

File: sdk/android/src/main/kotlin/com/contextsolutions/securegateway/mobile/SecureGateway.kt
package com.contextsolutions.securegateway.mobile

/**
 * Entry point for the mobile SDK — the single seam the host app toggles behind its relay
 * feature flag (PRD §8.1/§8.4). When off, the app keeps its legacy local QR-sync; when on,
 * it builds a [MobileClient] here. The QR is versioned, so a legacy QR (no `v`) falls back
 * to the old pairing path in the host app.
 */
object SecureGateway {

    fun mobile(config: MobileConfig): MobileClient = MobileClient(config)

    /** Kotlin DSL convenience: `SecureGateway.mobile { authUrl = ...; accountSecret = ... }`. */
    fun mobile(block: MobileConfig.() -> Unit): MobileClient =
        MobileClient(MobileConfig().apply(block))
}
